pkg/piperutils/slices_test.go

```diff
@@ -15,3 +16,14 @@ func TestContainsInt(t *testing.T) {
 	assert.Equal(t, true, ContainsInt(intList, 4), "True expected but returned false")
 	assert.Equal(t, false, ContainsInt(intList, 13), "False expected but returned true")
 }
+
+func TestPrefix(t *testing.T) {
+	// init
+	s := []string{"tree", "pie", "juice"}
+	// test
+	s = Prefix(s, "apple")
+	// assert
+	assert.Contains(t, s, "appletree")
+	assert.Contains(t, s, "applejuice")
+	assert.Contains(t, s, "applepie")
```

Comment:
There are no tests which reflect the additional side-effects. If someone stumbles over this and changes the Prefix behavior, it won't be revealed that this breaks sonar.